### PR TITLE
Persist state

### PR DESCRIPTION
**Pull request by aberaud**

### CODE REVIEW

**Comment by AmarOk1412** [comment]
(I'll review this tonight)

File: tools/tools_common.h [hunk 1]
```diff
@@ -154,6 +156,9 @@ parseArgs(int argc, char **argv) {
         case 'D':
             params.devicekey = optarg;
             break;
+        case 'f':
```
Comment (AmarOk1412) [inline tools/tools_common.h]: Missing "f" in getopt_long. (--persist is recognized, -f is not)

File: src/dht.cpp [hunk 2]
```diff
@@ -2424,4 +2431,52 @@ Dht::onAnnounceDone(const Sp<Node>& node, net::RequestAnswer& answer, Sp<Search>
...
+{
+    std::ofstream file(path);
+    msgpack::pack(file, exportNodes());
+    msgpack::pack(file, exportValues());
+}
+
+void
+Dht::loadState(const std::string& path)
+{
+    DHT_LOG.d("Importing state from %s", path.c_str());
```
Comment (AmarOk1412) [inline src/dht.cpp]: Note: these logs are not shown when using dhtnode.cpp, in fact loggers are enabled after `dht->run()` so loadState is already called.

Comment (aberaud) [reply src/dht.cpp]: Indeed.. I plan to add the logger as part of the dht constructor or config (might be in another patch)

Comment (AmarOk1412) [reply src/dht.cpp]: yup not related to this PR (that's why I just add a Note). I've just noticed this weird thing.

File: tools/tools_common.h (hunk 1 above)
Comment (aberaud) [reply tools/tools_common.h]: done

**Pull request merged**